Sources/RInAppMessaging/Extensions/UIColor+IAM.swift

```diff
@@ -26,4 +33,23 @@ internal extension UIColor {
         return brightness > threshold
     }
 
+    /// Returns distance between colors based on the low-cost approximation algorithm from [](https://www.compuphase.com/cmetric.htm)
+    func distance(from: UIColor) -> Int {
+        var (r1, g1, b1) = (CGFloat(0), CGFloat(0), CGFloat(0))
+        var (r2, g2, b2) = (CGFloat(0), CGFloat(0), CGFloat(0))
+        getRed(&r1, green: &g1, blue: &b1, alpha: nil) // works with RGB, HSB, extendedGray
+        from.getRed(&r2, green: &g2, blue: &b2, alpha: nil)
+
+        let rMean = (r1 * 255 + r2 * 255) / 2
+        let dR = r1 * 255 - r2 * 255
+        let dG = g1 * 255 - g2 * 255
+        let dB = b1 * 255 - b2 * 255
+
+        return Int(round(sqrt((2 + rMean / 256) * dR * dR + 4 * dG * dG + (2 + (255 - rMean) / 256) * dB * dB)))
```

Comment:
```c
return sqrt((((512+rmean)*r*r)>>8) + 4*g*g + (((767-rmean)*b*b)>>8));
```

The C interpretation in the paper used a (supposedly faster) bitshift right op by 8 instead of division in the original formula. Probably an academic thought in this case but wonder if there would've been a big perf difference

Reply:
Yeah, I decided to go with readability because this formula is not going to be often executed. So I think on modern devices and having Swift optimizer, there shouldn't be a big pref difference. I don't have any source to back me up on this though. If that was a C code I would use the bit shift approach 🙂